charm/kafka/reactive/kafka.py

```diff
@@ -63,10 +64,24 @@ def install_snap():
     snap_file = get_snap_file_from_charm() or hookenv.resource_get('kafka')
     if snap_file:
         hookenv.log('Detected Kafka snap, installing {}'.format(snap_file))
-        check_call(['snap', 'install', '--dangerous', snap_file])
+
+        if upgrade:
+            # Upgrading local snaps is not yet supported by snap, so we copy /var/snap/<kafka_snap>
+            with tempfile.TemporaryDirectory() as tmpdir:
+                shutil.copy(KAFKA_SNAP_DATA, tmpdir)
+                check_call(['snap', 'remove', KAFKA_SNAP])
```

Comment:
As mentioned at standup, removing may not be necessary, and not what we want to do here.

Doing a `snap install` of a snap from file over a prior install should work.